### PR TITLE
Added option for timeout while connecting to peripheral

### PR DESCRIPTION
**Pull request by GunnarI**
Added option to pass timeout parameter to the peripheral constructor to be used as timeout for "trying to connect" period.
Possible resolution for issue #303

### CODE REVIEW

**Comment by mdxs** [comment]
Thanks, I would really like to see this merged... for now I use a local MyPheripheral with this PR approach to get the functionality, which is extremely useful when trying to connect to devices that are (no longer) connectable. Without this PR the default timeout is too long for me.

File: docs/peripheral.rst [hunk 1]
```diff
@@ -24,6 +24,8 @@ Constructor
    The *iface* parameter allows the Bluetooth interface on which to make the connection to be set.
    On Linux, 0 means */dev/hci0*, 1 means */dev/hci1* and so on.
 
+   The *timeout* parameter (in seconds) can be used to limit the hang time for trying to connect to device
```
Comment (mdxs) [inline docs/peripheral.rst]: Add a dot "." at the end of the sentence.

**Comment by nikigraus** [comment]
I also need this change, otherwise I can't use the software as it is hanging too long when the device is not there.

**Comment by BaloMueller** [comment]
Also approving the pullrequest. It is working for us.

**Comment by sneko** [comment]
@IanHarvey also a good PR without breaking change if you plan a next release :)

Thank you,

**Pull request merged**